test/direct_consumer_test.exs

```diff
@@ -0,0 +1,93 @@
+defmodule DirectConsumerTest do
+  use ExUnit.Case
+
+  alias AMQP.{Basic, Connection, Channel, Queue}
+  alias Channel.ReceiverManager
+
+  setup do
+    {:ok, conn} = Connection.open()
+    {:ok, chan} = Channel.open(conn, {AMQP.DirectConsumer, self()})
```

Comment:
So this and the queue are slightly tricky, as when you call `Channel.open/2` with the direct consumer, the test process itself becomes the consumer, so that the channel is monitoring it. I can solve it by closing the channel, however that _cannot_ be done from the `on_exit` callback as by then the test process exited and the channel will complain. Let me see what I can do.